pkg/apis/core/validation/validation.go

```diff
@@ -5581,7 +5581,14 @@ func ValidatePodEphemeralContainersUpdate(newPod, oldPod *core.Pod, opts PodVali
 func ValidatePodResize(newPod, oldPod *core.Pod, opts PodValidationOptions) field.ErrorList {
 	// Part 1: Validate newPod's spec and updates to metadata
 	fldPath := field.NewPath("metadata")
-	allErrs := ValidateImmutableField(&newPod.ObjectMeta, &oldPod.ObjectMeta, fldPath)
```

Comment:
there are bigger issues here, I think... this will break if managed fields or generation updates ... 

rather than failing in validation if pod resize changes something in metadata, should we simply be resetting metadata (using ResetObjectMetaForStatus) in the resize prepareforupdate strategy method, then bumping generation?

Reply:
it looks like the strategy already calls dropNonResizeUpdates, which is close to what ResetObjectMetaForStatus does already... I don't think we need to validate objectmeta is immutable at ~all in this validation

Reply:
I found this old comment from @tallclair which says that it is dangerous to be copying the pod name/namespace/resourceVersion/uid without metadata validation. But I'm a little confused by that comment (what danger is there?) and the resolution, because metadata validation looks like it is happening _after_ the copying happens so I'm unclear how the validation prevents the aforementioned dangerous scenario from occurring

Reply:
all updates validate object metadata unconditionally and generically, independent of type-specific validation:

https://github.com/kubernetes/kubernetes/blob/72b404f6aadd55ec8c02e7045a91205016051a71/staging/src/k8s.io/apiserver/pkg/registry/rest/update.go#L148-L152

https://github.com/kubernetes/kubernetes/blob/72b404f6aadd55ec8c02e7045a91205016051a71/staging/src/k8s.io/apiserver/pkg/registry/rest/update.go#L87-L98

That ensures (among other things) that name / namespace / uid are not mutated, and resourceVersion is not emptied

Reply:
Makes sense. In that case, I agree that this metadata validation can be removed.

Reply:
took out the call to `ValidateImmutableField`, but I left the call to `ValidateObjectMetaUpdate` to keep it consistent with all the other validation functions here and also to validate the generation bump